ale_linters/erlang/erlc.vim

```diff
@@ -0,0 +1,88 @@
+" Author: Magnus Ottenklinger - https://github.com/evnu
+
+function! ale_linters#erlang#erlc#Handle(buffer, lines)
+  " Matches patterns like the following:
+  "
+  " error.erl:4: variable 'B' is unbound
+  " error.erl:3: Warning: function main/0 is unused
+  " error.erl:4: Warning: variable 'A' is unused
+  let l:pattern = '\v^([^:]+):(\d+): (Warning: )?(.+)$'
+
+  " parse_transforms are a special case. The error message does not indicate a location:
+  " error.erl: undefined parse transform 'some_parse_transform'
+  let l:pattern_parse_transform = '\v(undefined parse transform .*)$'
+  let l:output = []
+
+  let l:pattern_no_module_definition = '\v(no module definition)$'
+  let l:pattern_unused = '\v(.* is unused)$'
+
+  let l:is_hrl = expand('%:e') ==# 'hrl'
+
+  for l:line in a:lines
+    let l:match = matchlist(l:line, l:pattern)
+    let l:match_parse_transform = matchlist(l:line, l:pattern_parse_transform)
+
+    " Determine if the output indicates an error. We distinguish between two cases:
+    "
+    " 1) normal errors match l:pattern
+    " 2) parse_transform errors match l:pattern_parse_transform
+    "
+    " If none of the patterns above match, the line can be ignored
+    if len(l:match) == 0 " not a 'normal' warning or error
+        if len(l:match_parse_transform) == 0 " also not a parse_transform error
+            continue
+        endif
+
+        let l:text = l:match_parse_transform[0]
+        call add(l:output, {
+                    \   'bufnr': a:buffer,
```

Comment:
Use a handling indent for continuing lines, like in the other source files. So you only increase the whitespace by one level in indentation.